sdk/eidolon_ai_sdk/apu/llm/ollama_llm_unit.py

```diff
@@ -0,0 +1,165 @@
+import json
+import logging
+from typing import List, Optional, Union, Literal, Dict, Any, AsyncIterator, cast
+
+import yaml
+from fastapi import HTTPException
+from ollama import AsyncClient, ResponseError, Options
+
+from eidolon_ai_client.events import (
+    StringOutputEvent,
+    ObjectOutputEvent,
+)
+from eidolon_ai_client.util.logger import logger as eidolon_logger
+from eidolon_ai_sdk.apu.call_context import CallContext
+from eidolon_ai_sdk.apu.llm_message import (
+    LLMMessage,
+    AssistantMessage,
+    UserMessage,
+    SystemMessage,
+)
+from eidolon_ai_sdk.apu.llm_unit import LLMUnit, LLMCallFunction, LLMModel, LLMUnitSpec
+from eidolon_ai_sdk.system.reference_model import Specable, AnnotatedReference
+from eidolon_ai_sdk.util.replay import replayable
+
+logger = eidolon_logger.getChild("llm_unit")
+
+async def convert_to_ollama(message: LLMMessage):
+    if isinstance(message, SystemMessage):
+        return {"role": "system", "content": message.content}
+    elif isinstance(message, UserMessage):
+        content = message.content
+        if not isinstance(content, str):
+            content = ""
+            for part in message.content:
+                if part.type == "text":
+                    content += part.text
+                else:
+                    # not supported
+                    pass
+        return {"role": "user", "content": content}
+    elif isinstance(message, AssistantMessage):
+        # TODO: tool calls
```

Comment:
I assume you are supporting tools via the json wrapper, is this todo still relevant?

Reply:
No, i'll fix that and remove the commented out code below